workerextension.go

```
@@ -1,56 +1,30 @@
 package frankenphp
 
 import (
-	"errors"
 	"net/http"
 )
 
-// EXPERIMENTAL: Worker allows you to register a worker where, instead of calling FrankenPHP handlers on
-// frankenphp_handle_request(), the GetRequest method is called.
-//
-// You may provide an http.Request that will be conferred to the underlying worker script,
-// or custom parameters that will be passed to frankenphp_handle_request().
-//
-// After the execution of frankenphp_handle_request(), the return value WorkerRequest.AfterFunc will be called,
-// with the optional return value of the callback passed as parameter.
-//
-// A worker script with the provided name, fileName and thread count will be registered, along with additional
-// configuration through WorkerOptions.
-//
-// Workers are designed to run indefinitely and will be gracefully shut down when FrankenPHP shuts down.
-//
-// Extension workers receive the lowest priority when determining thread allocations. If MinThreads cannot be
-// allocated, then FrankenPHP will panic and provide this information to the user (who will need to allocate more
-// total threads). Don't be greedy.
-type Worker struct {
-	name     string
-	fileName string
-	num      int
-	options  []WorkerOption
+// EXPERIMENTAL: Workers allows you to register a worker.
+type Workers interface {
+	// SendRequest calls the closure passed to frankenphp_handle_request() and updates the context.
+	// The generated HTTP response will be written through the provided writer
+	SendRequest(rw http.ResponseWriter, r *http.Request) error
+	// SendMessage calls the closure passed to frankenphp_handle_request(), passes message as a parameter, and returns the value produced by the closure.
+	SendMessage(message any, rw http.ResponseWriter) any
```

I wonder if `SendMessage` should also return an error. It's not the case right now, but there should probably be a way to notify go if PHP fails

---

Or perhaps our php runtime crashed